Baby Monitor/Source Files/Modules/CameraPreview/CameraPreviewViewModel.swift

```diff
@@ -59,5 +65,12 @@ final class CameraPreviewViewModel {
                 self?.stop()
             })
             .disposed(by: bag)
+        socketCommunicationManager.communicationResetted
+            .throttle(1.0, scheduler: MainScheduler.instance)
```

Comment:
It depends on how quickly socket will produce connection error (eg. when server is killed, wifi is disconnected, etc.). Immediately after such an error, socket attempts to reestablish connection. It may take up to few seconds, but could be also faster. As setting up a stream might be costy operation, I decided to add at least a 1s pause between attempts.